src/js/profile/mobile/widget/DatePicker.js

```diff
@@ -0,0 +1,416 @@
+			* Init method
+			* @method _init
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._init = function () {
+				var self = this;
+
+				self._setValue(new Date());
+				// Update day in month
+				self._changeMonth(self._getValue("month"));
+
+				// Calculate the spin widget after building the widget
+				Object.keys(self._spins).forEach(function (key) {
+					self._spins[key].refresh();
+				});
+			};
+
+			/**
+			* Build widget for the date picker
+			* @method _buildDatePicker
+			* @param {string} name
+			* @return {HTMLElement} container for date spin
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._buildDatePicker = function (name) {
+				var self = this,
+					ui = self._ui,
+					spin = document.createElement("div"),
+					spinContainer = document.createElement("div"),
+					options = {},
+					spinWidget;
+
+				spin.classList.add(Spin.classes.SPIN);
+				spinContainer.classList.add(DatePicker.classes.CONTAINER);
+
+				if (name === "day") {
+					options = {
+						min: 1,
+						max: 31
+					};
+					spinContainer.classList.add(DatePicker.classes.DAY_CONTAINER);
+				} else if (name === "month") {
+					options = {
+						min: 1,
+						max: MONTH_NAMES.length,
+						labels: MONTH_NAMES.map(function (value) {
+							return value.toUpperCase().substring(0, 3);
+						}).join(",")
+					};
+					spinContainer.classList.add(DatePicker.classes.MONTH_CONTAINER);
+				} else if (name === "year") {
+					options = {
+						min: MIN_YEAR,
+						max: MAX_YEAR,
+						value: 2020
+					}
+					spinContainer.classList.add(DatePicker.classes.YEAR_CONTAINER);
+				}
+
+				spinContainer.appendChild(spin);
+
+				ui[name + "Spin"] = spin;
+
+				spinWidget = ns.widget.Spin(spin, options);
+				self._spins[name] = spinWidget;
+
+				return spinContainer;
+			};
+
+			/**
+			* Build widget instance
+			* @method _build
+			* @param {HTMLElement} element
+			* @return {HTMLElement} Builded element
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._build = function (element) {
+				var self = this,
+					ui = self._ui,
+					header = document.createElement("div"),
+					content = document.createElement("div"),
+					dayPicker = self._buildDatePicker("day"),
+					monthPicker = self._buildDatePicker("month"),
+					yearPicker = self._buildDatePicker("year");
+
+				header.classList.add(DatePicker.classes.HEADER);
+				element.appendChild(header);
+
+				ui.header = header;
+
+				content.classList.add(DatePicker.classes.CONTENT);
+				content.appendChild(dayPicker);
+				content.appendChild(monthPicker);
+				content.appendChild(yearPicker);
+
+				element.appendChild(content);
+
+				return element;
+			};
+
+			/**
+			* Return Date value or value of one field
+			* @method _getValue
+			* @param {string} type
+			* @return {Date|number}
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._getValue = function (type) {
```

Comment:
this method cannot take an argument... Please check the BaseWidget.value method.

Method .value() allows two cases:
- without argument - get value
- with argument - set value

Reply:
`src/js/profile/wearable/widget/wearable/DatePicker.js` also seems to need a fix.
Okay, I will add a new function to separate it.